### PR TITLE
Document machine readable json output for dotnet list package report

### LINKED ISSUE #2922: Create dotnet list package json output command document
Follow up on: https://github.com/NuGet/NuGet.Client/pull/4855
Create MS NuGet doc page and blog post.

### PR DESCRIPTION
**Pull request by erdembayar**
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/2922
Related to: https://github.com/dotnet/sdk/pull/29367, https://github.com/NuGet/Home/issues/7752

Spec is [here](https://github.com/NuGet/Home/blob/557f5dfdcdaf21f7a83dab6d30a35d05d33d489b/proposed/2022/DotnetListPackageMachineReadableJsonOutput.md). New command option can be used like below:
`dotnet list package --format json`
`dotnet list package --format json --output-version 1`

**!!! Don't merge before we verify it's available in GA sdk version.**

### CODE REVIEW

**Review by nkolev92 — APPROVED** [review]
Do we want to document the format somehow/somewhere?

**Comment by erdembayar** [comment]
> Do we want to document the format somehow/somewhere?

Do you mean something like blog post which explains usage in detail?

**Comment by nkolev92** [comment]
> > Do we want to document the format somehow/somewhere?
> 
> Do you mean something like blog post which explains usage in detail?

Not really. More of a schema for the file somewhere that's not our designs. 
People would need to know how to parse things.

File: docs/core/tools/dotnet-list-package.md [hunk 1]
```diff
@@ -118,6 +120,14 @@ The project or solution file to operate on. If not specified, the command search
 
   Lists packages that have known vulnerabilities. Cannot be combined with `--deprecated` or `--outdated` options. Nuget.org is the source of information about vulnerabilities. For more information, see [Vulnerabilities](/nuget/api/registration-base-url-resource) and [How to Scan NuGet Packages for Security Vulnerabilities](https://devblogs.microsoft.com/nuget/how-to-scan-nuget-packages-for-security-vulnerabilities/).
 
+- **`--format <console|json>`**
+
+  Sets the report output format. Allowed values are `console`, `json`.  Defaults to `console`.
+
+- **`--output-version <VERSION>`**
+
+  Sets the report output version. Allowed value is `1`. Defaults to `1`. Requires the `--format json` option.
```
Comment (tdykstra) [inline docs/core/tools/dotnet-list-package.md:129]:
```suggestion
  Sets the report output version. Allowed value is `1`. Defaults to `1`. Requires the `--format json` option. When a new JSON version is available, the command will produce the new format by default. This option will let you specify that the command should produce an earlier format.
```

**Review by tdykstra — APPROVED** [review]
LGTM, one suggestion

File: docs/core/tools/dotnet-list-package.md [hunk 2]
```diff
@@ -137,3 +147,27 @@ The project or solution file to operate on. If not specified, the command search
...
+- List package references for a specific target framework in machine readable json output format:
+
+  ```dotnetcli
+   dotnet list package --framework netcoreapp3.0 --format json
+  ```
+
+- Save machine readable json output of package references, including transitive dependency and vulnerability details into a file:
+
+  ```dotnetcli
+  dotnet list package --include-transitive --vulnerable --format json >> dependencyReport.json
```
[the remark this answers is not in the document]
Comment (erdembayar) [reply docs/core/tools/dotnet-list-package.md:166]: I don't know. For now, I'm assuming it's working on CI pipeline, so `dependencyReport.json` doesn't exist when command runs.

**Comment by tdykstra** [comment]
> > > Do we want to document the format somehow/somewhere?
> > 
> > 
> > Do you mean something like blog post which explains usage in detail?
> 
> Not really. More of a schema for the file somewhere that's not our designs. People would need to know how to parse things.

@nkolev92 @erdembayar We have an issue now asking for the format (#38022) is that information available somewhere?

**Comment by nkolev92** [comment]
We have a tracking issue for that (it was internal, so I moved it to an external one) https://github.com/NuGet/docs.microsoft.com-nuget/issues/3184. 

The source of truth is https://github.com/NuGet/Home/blob/dev/proposed/2022/DotnetListPackageMachineReadableJsonOutput.md. 
It's not quite written ina  schema way, but it contains all the examples.

**Comment by Marusyk** [comment]
@nkolev92 
https://github.com/NuGet/Home/blob/dev/proposed/2022/DotnetListPackageMachineReadableJsonOutput.md
image

**Comment by erdembayar** [comment]
> @nkolev92 https://github.com/NuGet/Home/blob/dev/proposed/2022/DotnetListPackageMachineReadableJsonOutput.md image

Someone recently moved the specs, use new link: https://github.com/NuGet/Home/blob/dev/accepted/2022/DotnetListPackageMachineReadableJsonOutput.md

**Pull request merged**